### PR TITLE
Add ruff workflow

### PR DESCRIPTION
**Pull request by atharva-2001**
### :pencil: Description

Adds a ruff job alongside black. 

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: .github/workflows/codestyle.yml [hunk 1]
```diff
@@ -29,3 +33,74 @@ jobs:
...
       - name: Run Black
         run: black --check tardis
+  ruff: 
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
```
Comment (Knights-Templars) [inline .github/workflows/codestyle.yml:43]: Is the python version up to date?

File: .github/workflows/codestyle.yml [hunk 2]
```diff
@@ -29,3 +33,74 @@ jobs:
...
+              
+            </details>
+        env:
+          URL: https://github.com/${{ github.repository_owner }}/${{ github.event.repository.name }}/actions/runs/${{ github.run_id }}?check_suite_focus=true
+          
+      - name: Save results artifact
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: asv-benchmark-results-${{ runner.os }}
```
Comment (KasukabeDefenceForce) [inline .github/workflows/codestyle.yml:102]: I think the name should be different form benchmark.

File: .github/workflows/codestyle.yml [hunk 3]
```diff
@@ -29,3 +33,74 @@ jobs:
...
       - name: Run Black
         run: black --check tardis
+  ruff: 
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12.3'
```
Comment (andrewfullard) [inline .github/workflows/codestyle.yml:43]: Do we keep this in track with TARDIS?

File: .github/workflows/codestyle.yml [hunk 4]
```diff
@@ -29,3 +33,74 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12.3' 
+      
+      - name: Install Ruff
+        run: pip install ruff
```
Comment (andrewfullard) [inline .github/workflows/codestyle.yml:46]: We should specify the ruff version

**Review by andrewfullard — CHANGES_REQUESTED** [review]

File: .github/workflows/codestyle.yml (hunk 3 above)
Comment (atharva-2001) [reply .github/workflows/codestyle.yml:43]: Yes,
https://github.com/tardis-sn/tardis/blob/0df98ce0c9bf392c3e1b2406c27b4faa6ad07060/conda-lock.yml#L6522

File: .github/workflows/codestyle.yml (hunk 4 above)
Comment (atharva-2001) [reply .github/workflows/codestyle.yml]: I used the version from the lock file.

**Review by andrewfullard — DISMISSED** [review]

File: .github/workflows/codestyle.yml [hunk 5]
```diff
@@ -29,3 +33,74 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12.3' 
+      
+      - name: Install Ruff
+        run: pip install ruff==0.4.9
```
Comment (wkerzendorf) [inline .github/workflows/codestyle.yml:46]: update it.

Comment (atharva-2001) [reply .github/workflows/codestyle.yml]: done in [`fea7705` (#2660)](https://github.com/tardis-sn/tardis/pull/2660/commits/fea77053fd38874079a558034e15c96240638210)

**Review by KasukabeDefenceForce — APPROVED** [review]

**Pull request merged**